usb.go

```diff
@@ -210,6 +210,36 @@ func (c *Context) OpenDevices(opener func(desc *DeviceDesc) bool) ([]*Device, er
 	return ret, reterr
 }
 
+func (c *Context) OpenDeviceWithFileDescriptor(fd uintptr) (*Device, error) {
+	handle, err := c.libusb.wrapSysDevice(c.ctx, fd)
+	if err != nil {
+		return nil, err
+	}
+
+	dev, err := c.libusb.getDevice(handle)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := c.libusb.getDeviceDesc(dev)
+	defer c.libusb.dereference(dev)
```

Comment:
I don't think this is needed - according to the docs, libusb_get_device doesn't increase reference count and dereferencing is not required.